include/tvm/utils/internal/MapWithVariableAsKey.h

```diff
@@ -0,0 +1,96 @@
+/** Copyright 2017-2020 CNRS-AIST JRL and CNRS-UM LIRMM */
+
+#pragma once
+
+#include <tvm/Variable.h>
+
+#include <tvm/utils/internal/map.h>
+
+namespace tvm::utils::internal
+{
+/** Dummy struct for selecting overload*/
+struct with_sub
+{};
+
+template<typename T>
+struct slice_traits
+{};
+
+/** A map with Variables as key, that can handle the notion of subvariables.*/
```

Comment:
This should indicate what's the `Slicer` template type and what it supposed to implement and why

Reply:
Done in 072db5478a270c4c72684cf39909cb29dfb4d8f0